sensors/alertmanager_sensor.py

```diff
@@ -0,0 +1,61 @@
+from st2reactor.sensor.base import Sensor
+
+import BaseHTTPServer
+from BaseHTTPServer import BaseHTTPRequestHandler
+
+import urllib
+import json
+import traceback
+
+class AlertmanagerSensor(Sensor):
+  def __init__(self, sensor_service, config):
+    super(AlertmanagerSensor, self).__init__(sensor_service=sensor_service, config=config)
+    self._logger = self.sensor_service.get_logger(name=self.__class__.__name__)
+    self._host = '0.0.0.0'
+    self._port = 12345
+
+  def _create_handler_class(sensor):
+    class AlertmanagerHandler(BaseHTTPRequestHandler):
```

Comment:
I would just pull this `class` up.  It's cool enough to stand on it's own.  Then you would just call the constructor in `setup()` instead of having a special function.  You would also need want to change your constructor to take `sensor`.

Reply:
So originally that's what I did. However, my program isn't what calls the constructor. I just pass the class to `BaseHTTPServer.HTTPServer` and that's what ends up constructing the `AlertmanagerHandler` object. So Stack Exchange told me this was how I should go about it. :)

I suppose I could have peaked into `BaseHTTPServer.HTTPServer`, subclassed it, and overridden the method that calls the constructor for `AlertmanagerHandler`, but that seemed less clean.